### PR TITLE
added anti knockback abilities

### PR DESCRIPTION
**Pull request by CarnifexOptimus**
Added 3 anti knockback abilities to the knockback component. I hope I have done this right and also not forget about any.

### CODE REVIEW

File: BossMod/Components/Knockback.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@ protected struct PlayerImmuneState
         {
             public DateTime ArmsLengthSurecastExpire; // 0 if not active
             public DateTime InnerStrengthExpire; // 0 if not active
+            public DateTime ManawallExpire; // 0 if not active
 
             public bool ImmuneAt(DateTime time) => ArmsLengthSurecastExpire > time || InnerStrengthExpire > time;
```
Comment (awgil) [inline BossMod/Components/Knockback.cs:35]: You also need to include the check here for it to actually work.

File: BossMod/Components/Knockback.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@ protected struct PlayerImmuneState
         {
             public DateTime ArmsLengthSurecastExpire; // 0 if not active
             public DateTime InnerStrengthExpire; // 0 if not active
+            public DateTime ManawallExpire; // 0 if not active
```
Comment (awgil) [inline BossMod/Components/Knockback.cs:33]: Maybe rename to Role/Job/Duty?

Comment (CarnifexOptimus) [reply BossMod/Components/Knockback.cs:33]: good idea

File: BossMod/Components/Knockback.cs (hunk 1 above)
Comment (CarnifexOptimus) [reply BossMod/Components/Knockback.cs]: fixed it

**Pull request merged**